api/tests/lib/db_testing.py

```diff
@@ -18,21 +19,27 @@ def create_isolated_db(monkeypatch) -> db.DBClient:
     that connects to that schema. Drops the schema after the context manager
     exits.
     """
-    schema_name = f"test_schema_{uuid.uuid4().int}"
-    monkeypatch.setenv("DB_SCHEMA", schema_name)
+
+    schema_prefix = f"test_{uuid.uuid4().int}_"
+
+    test_schemas = [f"{schema_prefix}{schema}" for schema in Schemas]
+
+    monkeypatch.setenv("SCHEMA_PREFIX_OVERRIDE", schema_prefix)
     monkeypatch.setenv("DB_CHECK_CONNECTION_ON_INIT", "False")
 
     # To improve test performance, don't check the database connection
     # when initializing the DB client.
     db_client = db.PostgresDBClient()
```

Comment:
IMHO a slightly cleaner solution without monkeypatching would be to build a `PostgresDBConfig` directly and then pass it to `PostgresDBClient()`. Something like this:

```suggestion
    db_config = PostgresDBConfig(schema_prefix_override=schema_prefix)

    monkeypatch.setenv("DB_CHECK_CONNECTION_ON_INIT", "False")

    # To improve test performance, don't check the database connection
    # when initializing the DB client.
    db_client = db.PostgresDBClient(db_config)
```

Reply:
I can reconfigure the tests to setup in the way you're suggesting, but we still require the `    monkeypatch.setenv("SCHEMA_PREFIX_OVERRIDE", db_schema_prefix)` line. We need to set that so the postgres db config that the app client initializes also has the prefix so that it can also set up its schema translate map correctly.